controls/models.py

```diff
@@ -205,14 +210,44 @@ def assign_baseline_controls(self, user, baselines_key, baseline_name):
             # print("User does not have permission to assign selected controls to element's system.")
             return False
 
+    def statements(self, statement_type):
+        """Return on the statements of statement_type produced by this element"""
+        smts = Statement.objects.filter(producer_element = self, statement_type = statement_type)
+        return smts
+
+    @transaction.atomic
+    def copy(self, name=None):
+        """Return a copy of an existing system element as a new element with duplicate control_implementation_prototype statements"""
+
+        # Copy only elements that are components. Do not copy an element of type "system"
+        # Components that are systems should always be associated with a project (at least currently).
+        # Also, statement structure for a system would be very different.
+        if self.element_type == "system":
+            raise SystemException("Copying an entire system is not permitted.")
+
+        e_copy = deepcopy(self)
+        e_copy.id = None
+        if name is not None:
+            e_copy.name = name
+        else:
+            e_copy.name = self.name + " copy"
```

Comment:
Sets you up nicely to add the edit-name feature later on, good planning for this function.